server/src/test/java/org/elasticsearch/snapshots/CorruptedBlobStoreRepositoryIT.java

```diff
@@ -356,4 +367,9 @@ private void assertRepositoryBlocked(Client client, String repo, String existing
         assertThat(repositoryException4.getMessage(),
             containsString("Could not read repository data because the contents of the repository do not match its expected state."));
     }
+
+    private void fullRestart() throws Exception {
```

Comment:
This is admittedly pretty dirty now test-wise. Obviously, it also illustrates that this change slightly weakens our ability to detect concurrent repository writes from multiple clusters.
IMO, this is a fair trade-off though and one we've already been making over and over in other spots (anytime we optimised away a loading of `RepositoryData` we weakened the ability to detect concurrent load obviously).